scripts/zones/Port_Jeuno/npcs/Squintrox_Dryeyes.lua

```diff
@@ -3,117 +3,534 @@
+                player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.SEEDSPALL_ROSEUM)
+            end
+            if not player:hasKeyItem(xi.ki.SEEDSPALL_CAERULUM) then
+                player:addKeyItem(xi.ki.SEEDSPALL_CAERULUM)
+                player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.SEEDSPALL_CAERULUM)
+            end
+            if not player:hasKeyItem(xi.ki.SEEDSPALL_VIRIDIS) then
+                player:addKeyItem(xi.ki.SEEDSPALL_VIRIDIS)
+                player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.SEEDSPALL_VIRIDIS)
+            end
+        elseif option == 301 then -- Mark of Seed
+            player:delSeals(15, 0)
+            player:delGil(1500)
+
+            if not player:hasKeyItem(xi.ki.MARK_OF_SEED) then
+                player:addKeyItem(xi.ki.MARK_OF_SEED)
+                player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.MARK_OF_SEED)
+            end
+        elseif option == 302 then -- Omnis Stone
+            player:delSeals(20, 0)
+            player:delGil(2000)
+
+            if not player:hasKeyItem(xi.ki.OMNIS_STONE) then
+                player:addKeyItem(xi.ki.OMNIS_STONE)
+                player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.OMNIS_STONE)
+            end
+        elseif option == 303 then -- Cardians mana orbs
+            player:delSeals(5, 0)
+            player:delGil(500)
+
+            for i, ki in pairs({
+                    xi.ki.ORB_OF_CUPS,
+                    xi.ki.ORB_OF_COINS,
+                    xi.ki.ORB_OF_BATONS,
+                    xi.ki.ORB_OF_SWORDS,
+                }) do
+                if not player:hasKeyItem(ki) then
+                    player:addKeyItem(ki)
+                    player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+                end
+            end
+        elseif option == 304 then -- Navaratna Talisman
+            player:delSeals(15, 0)
+            player:delGil(1500)
+
+            local ki = xi.ki.NAVARATNA_TALISMAN
+            if not player:hasKeyItem(ki) then
+                player:addKeyItem(ki)
+                player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+            end
+        elseif option == 305 then -- Mega Bonanza Kupon
+            player:delSeals(20, 0)
+            player:delGil(2000)
+
+            local ki = xi.ki.MEGA_BONANZA_KUPON
+            if not player:hasKeyItem(ki) then
+                player:addKeyItem(ki)
+                player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+            end
+        elseif option == 306 then -- Black Book
+            player:delSeals(5, 0)
+            player:delGil(500)
+
+            local ki = xi.ki.BLACK_BOOK
+            if not player:hasKeyItem(ki) then
+                player:addKeyItem(ki)
+                player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+            end
+        elseif option == 307 then -- Elemental Saps
+            player:delSeals(15, 0)
+            player:delGil(1500)
+
+            for i, ki in pairs({
```

Comment:
For things like this, I'd define the table first and then use it, rather than inline in the loop definition

Reply:
I like it